src/QMCHamiltonians/OperatorBase.h

```diff
@@ -108,28 +111,21 @@ class OperatorBase : public QMCTraits
   };
 
   ///set the current update mode
-  std::bitset<8> UpdateMode;
+  std::bitset<8> update_mode;
```

Comment:
since you are touching variable names. Please add underscore to all member variables which has been touched.

Reply:
@ye-luo quick question, in the docs it says only protected and private, see [here](https://qmcpack.readthedocs.io/en/develop/developing.html#class-data-members). Should this also apply to public members? I'm only touching those 'cause it's already touching many files.

Reply:
I went through the manual and found
```
Class private/protected data members names should follow the convention of variable names with a trailing underscore (_).
```
@PDoakORNL do you remember why there is no mention of public members? Is it because we do not expect public members? Since there are public members from existing code, should we just request public members to carry underscore as well?

Reply:
Probably the comment about _ on public members was missed since there really shouldn't be any?

Reply:
It seems that renaming anyway changes the consumer side, I think we should put them as protected and then add const accessor for the consumer.

Reply:
How do we want to handle POD struct public members?

Reply:
> How do we want to handle POD struct public members?

I think POD struct doesn't need underscore.
But if a struct has member functions, we should use class instead of struct and also minimize public members.

Reply:
Echoing Ye's comment: nearly all the structs should be classes. And the public access to their data should be removed and go via accessors. There are a lot of these so it is something for us all to chip away at as we come across them.

Reply:
Sound good. Would the exception "global" read-only such as `enum class` and `const/constexpr static`? See those proposed in PR #3412 for [enum class](https://github.com/williamfgc/qmcpack/blob/refactoring-OperatorBase/src/QMCHamiltonians/OperatorBase.h#L90) and [static constexpr](https://github.com/williamfgc/qmcpack/blob/refactoring-OperatorBase/src/QMCHamiltonians/OperatorBase.h#L90).

Reply:
If the spirit of an object is still that of a data structure then public access to data members is fine. But member functions should not do "computations" and it should be strongly considered whether a pure function might be more appropriate.

Reply:
To be clear, I don't think there should be any public data members in a class excepting consts which I don't think should have an underscore. (I have no good reason for that). I don't count type aliases in this, they are just named as types whether they are public or private in a class. 

Currently I put _ onto existing data members if I keep them as I move them to protected or private.

Reply:
@williamfgc So in summary,
when a class data member shows up as public, it is already a red flag. We'd like to have no pubic class member except compile time constants or types.
Then there will be no question about whether to add underscore or not on public data member.

POD struct has its own use. it is meant to be all public. So we don't need to add underscore.

The purpose of adding underscore to easily identify whether a variable is internal or local in a member function.
```
foo::boo()
var_name // must be local or argument
var_name_ // internal
```

Reply:
Could add a line in the doc explaining why public is not listed?
"Rethink why using a public data member at the first place." something similar.

So in this PR, I think it will be good to use protected.

Reply:
We should also think about how to take advantage of clang-tidy.

Reply:
Sounds like a plan, thanks. Also, do we want to standardize accessor functions with a `getcX` and `getX` pattern? Something similar to what it's outlined in the [docs](https://qmcpack.readthedocs.io/en/develop/developing.html#class-format) and also how iterators distinguish between `const` and `non-const`.

```
const X& getcX() const; // read-only reference
X& getX(); // expect to modify member
```
Using templates is another option `get<T>()`, but would have the same effect (`getX ~ get<X>` if X is registered).
It would make things "findable" since when are trying to make code [FAIR](https://www.nature.com/articles/sdata201618) data.  Getting philosophical, but it's good to address this.

Reply:
If a member data needs to be accessed outside the class, then we provide const reference accessor. Only if a non-const reference accessor is really really necessary, we provide it and its name should be the same as the const one. For example "std::vector::data()" has both const and non-const version.

I don't get the second part of your comment about template. X is a data member instead of data type.

Reply:
@ye-luo OK, we don't foresee much use of non-const access, that's good. Agree with `std::vector::data()` example, iterators certainly took a separate semantics route.

> I don't get the second part of your comment about template. X is a data member instead of data type.

with C++ `auto template` (since C++17) we can pass enum entries as types (the member would be `X_`, while `X` is a enum that lists publicly accessed variables), I'm doing something like that [here](https://github.com/ORNL/ncio/blob/main/bindings/CXX17/ncio/cxx17/cxx17DataDescriptor.tcc#L113), the type is captured with `decltype` for completeness. Probably, the amount of boilerplate doesn't justify for this case as getX and get<X> are equally findable in any IDE or editor (getX is simpler) which is one of the goals.

Reply:
@ye-luo nevermind, C++17 `T& get <auto, T>` T recognition doesn't play well with return functions (the example shown is with `void`), not to mention consumer `lvalues` can be `auto` making even more confusing to the compiler. Nothing to win, but I learned something new today. Thanks.